### PR TITLE
text(ingestion): test multiple python versions in CI

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -47,17 +47,31 @@ jobs:
...
       - name: Install dependencies
         run: ./metadata-ingestion/scripts/install_deps.sh
       - name: Run metadata-ingestion tests
-        run: ./gradlew :metadata-ingestion:build :metadata-ingestion:check :metadata-ingestion:testFull
+        run: ./gradlew :metadata-ingestion:build :metadata-ingestion:check
+  metadata-ingestion-by-version:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.9"]
```
Comment (kevinhu) [inline .github/workflows/build-and-test.yml:65]: Why not 3.7 and 3.8?

**Review by kevinhu — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (hsheth2) [reply .github/workflows/build-and-test.yml:65]: I figured that testing 3.6 (min supported version) and 3.9 (latest released version) would be sufficient. We definitely could add more versions to test if we feel it necessary, but I don't think we're really at risk here.

Comment (kevinhu) [reply .github/workflows/build-and-test.yml:65]: Sounds good!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**